### PR TITLE
Adding documentation for the Debezium Spanner connector

### PR DESCRIPTION
**Pull request by nancyxu123**

### CODE REVIEW

File: documentation/modules/ROOT/pages/connectors/spanner.adoc [hunk 1]
```diff
@@ -0,0 +1,1201 @@
...
+|The maximum buffer of the queue in bytes used to pass events between the streamer and the main Kafka Connect loop.
+
+|[[connectors-strm-metric-currentqueuesizeinbytes]]<<connectors-strm-metric-currentqueuesizeinbytes, `+CurrentQueueSizeInBytes+`>>
+|`long`
+|The current buffer of the queue in bytes used to pass events between the streamer and the main Kafka Connect loop.
+
+|[[connectors-strm-metric-lowwatermark]]<<connectors-strm-metric-lowwatermark, `+LowWatermark+`>>
+|`long`
+|The current low watermark of the connector task. The low watermark describes the time T at which the connector is guaranteed to have streamed out all events with timestamp < T.
+
```
Comment (jpechane) [inline documentation/modules/ROOT/pages/connectors/spanner.adoc:974]: `milliSecondsLowWatermark` and `milliSecondsLowWatermarkLag` is missing

File: documentation/modules/ROOT/pages/connectors/spanner.adoc [hunk 2]
```diff
@@ -0,0 +1,1201 @@
...
+|[[connectors-strm-metric-latencylowwatermarklag]]<<connectors-strm-metric-latencylowwatermarklag, `+LatencyLowWatermarkLag+`>>
```
Comment (jpechane) [inline documentation/modules/ROOT/pages/connectors/spanner.adoc:975]: Please corect the metric name, should be like `latencyLowWatermarkLag<variant>MilliSeconds`. Please not the basic name was not correct and I introduced a mask-format where `<variant>` will be the P50 etc.
Same issue is for the following metics.

File: documentation/modules/ROOT/pages/connectors/spanner.adoc [hunk 3]
```diff
@@ -0,0 +1,1201 @@
...
+|`long`
+|The Spanner commit timestamp to Kafka publish timestamp latency. P50, P95, P99, Average, Min, Max metrics are provided.
+
+|[[connectors-strm-metric-latencyemittopublish]]<<connectors-strm-metric-latencyemittopublish, `+LatencyEmitToPublish+`>>
+|`long`
+|The connector emit timestamp to Kafka publish timestamp latency. P50, P95, P99, Average, Min, Max metrics are provided.
+
+|[[connectors-strm-metric-spannereventqueuecapacity]]<<connectors-strm-metric-spannereventqueuecapacity, `+SpannerEventQueueCapacity+`>>
+|`long`
+|The total Spanner event queue capacity.
```
Comment (jpechane) [inline documentation/modules/ROOT/pages/connectors/spanner.adoc:1001]: Explanation for Spanner event queue and how it differes from the regular above is missing.

**Review by jpechane — CHANGES_REQUESTED** [review]
@nancyxu123 Review done, thanks! I

- force-pushed to your branch to prpovide correct commit message
- fixed formatting quirks and missing changes in navigation
- added an icubation note

The text itself LGTM from the content point of view. I left few comments where I have doubts so please check and correct if needed.

File: documentation/modules/ROOT/pages/connectors/spanner.adoc [hunk 4]
```diff
@@ -0,0 +1,1201 @@
...
+
+The following skeleton JSON shows the basic four parts of a change event. However, how you configure the Kafka Connect converter that you choose to use in your application determines the representation of these four parts in change events. A `schema` field is in a change event only when you configure the converter to produce it. Likewise, the event key and event payload are in a change event only if you configure a converter to produce it. If you use the JSON converter and you configure it to produce all four basic change event parts, change events have this structure:
```
[the remark this answers is not in the document]
Comment (nancyxu123) [reply documentation/modules/ROOT/pages/connectors/spanner.adoc:105]: Got it. I changed this section to make it clear that change event key and change event volue are separate documents.

File: documentation/modules/ROOT/pages/connectors/spanner.adoc [hunk 5]
```diff
@@ -0,0 +1,1201 @@
...
+* If the event was part of a snapshot
+* The record sequence number for the data change event in the transaction
+* The project ID
+* The instance ID
+* The database ID
+* The change stream name
+* The transaction ID    
+* The low watermark, which represents the timestamp at which all records with commit timestamp less than the low watermark timestamp have already been streamed out by the connector
+* The commit timestamp for when the change was made in the database
+* The timestamp at which the connnector processed the change
```
[the remark this answers is not in the document]
Comment (nancyxu123) [reply documentation/modules/ROOT/pages/connectors/spanner.adoc:540]: Yes, that is the read_at_timestamp field

File: documentation/modules/ROOT/pages/connectors/spanner.adoc (hunk 2 above)
Comment (nancyxu123) [reply documentation/modules/ROOT/pages/connectors/spanner.adoc:975]: Sure, to be clear, are you asking me to rename it to: latencyLowWatermarkLag<variant>MilliSeconds?

And I will specify in the comments that variant can be P50, P95, P99, etc.

**Pull request merged**